[Bug] [Task plugin] datax task error

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

[LOG-PATH]: /opt/dolphinscheduler/worker-server/logs/20221014/7195111179040_2-15-22.log, [HOST]:  Host{address='172.16.10.15:1234', ip='172.16.10.15', port=1234}
[INFO] 2022-10-14 08:44:09.886 +0800 - Begin to pulling task
[INFO] 2022-10-14 08:44:09.887 +0800 - Begin to initialize task
[INFO] 2022-10-14 08:44:09.887 +0800 - Set task startTime: Fri Oct 14 08:44:09 CST 2022
[INFO] 2022-10-14 08:44:09.887 +0800 - Set task envFile: /opt/dolphinscheduler/worker-server/conf/dolphinscheduler_env.sh
[INFO] 2022-10-14 08:44:09.887 +0800 - Set task appId: 15_22
[INFO] 2022-10-14 08:44:09.887 +0800 - End initialize task
[INFO] 2022-10-14 08:44:09.888 +0800 - Set task status to TaskExecutionStatus{code=1, desc='running'}
[INFO] 2022-10-14 08:44:09.888 +0800 - TenantCode:root check success
[INFO] 2022-10-14 08:44:09.888 +0800 - ProcessExecDir:/tmp/dolphinscheduler/exec/process/7193666667040/7195111179040_2/15/22 check success
[INFO] 2022-10-14 08:44:09.888 +0800 - Resources:{} check success
[INFO] 2022-10-14 08:44:09.889 +0800 - Task plugin: DATAX create success
[INFO] 2022-10-14 08:44:09.889 +0800 - datax task params {"localParams":[],"resourceList":[],"customConfig":0,"dsType":"MYSQL","dataSource":1,"dtType":"MYSQL","dataTarget":1,"sql":"select id,name from ods.ods_jdx_site","targetTable":"ods_jdx_site_copy1","jobSpeedByte":0,"jobSpeedRecord":1000,"preStatements":[],"postStatements":[],"xms":1,"xmx":1}
[INFO] 2022-10-14 08:44:09.889 +0800 - Success initialized task plugin instance success
[INFO] 2022-10-14 08:44:09.889 +0800 - Success set taskVarPool: null
[ERROR] 2022-10-14 08:44:09.890 +0800 - datax task error
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.addCustomParameters(DataxTask.java:426)
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.buildShellCommandFile(DataxTask.java:400)
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.handle(DataxTask.java:157)
	at org.apache.dolphinscheduler.server.worker.runner.DefaultWorkerDelayTaskExecuteRunnable.executeTask(DefaultWorkerDelayTaskExecuteRunnable.java:48)
	at org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable.run(WorkerTaskExecuteRunnable.java:151)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[ERROR] 2022-10-14 08:44:09.890 +0800 - Task execute failed, due to meet an exception
org.apache.dolphinscheduler.plugin.task.api.TaskException: Execute DataX task failed
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.handle(DataxTask.java:171)
	at org.apache.dolphinscheduler.server.worker.runner.DefaultWorkerDelayTaskExecuteRunnable.executeTask(DefaultWorkerDelayTaskExecuteRunnable.java:48)
	at org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable.run(WorkerTaskExecuteRunnable.java:151)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.addCustomParameters(DataxTask.java:426)
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.buildShellCommandFile(DataxTask.java:400)
	at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.handle(DataxTask.java:157)
	... 9 common frames omitted
[INFO] 2022-10-14 08:44:10.900 +0800 - Get a exception when execute the task, will send the task execute result to master, the current task execute result is TaskExecutionStatus{code=6, desc='failure'}

### What you expected to happen

task run success

### How to reproduce

datax task with mysql2mysql

### Anything else

DATAX_HOME=/opt/datax

### Version

3.1.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

@DarkAssassinator the following is worker log while running datax

> [INFO] 2022-10-17 09:27:15.486 +0800 org.apache.dolphinscheduler.server.log.LoggerRequestProcessor:[79] - [WorkflowInstance-0][TaskInstance-0] - received command : Command [type=ROLL_VIEW_LOG_REQUEST, opaque=169, bodyLen=117]
[INFO] 2022-10-17 09:27:15.533 +0800 org.apache.dolphinscheduler.server.log.LoggerRequestProcessor:[79] - [WorkflowInstance-0][TaskInstance-0] - received command : Command [type=ROLL_VIEW_LOG_REQUEST, opaque=171, bodyLen=118]
[INFO] 2022-10-17 09:27:19.613 +0800 org.apache.dolphinscheduler.server.worker.task.WorkerHeartBeatTask:[88] - [WorkflowInstance-0][TaskInstance-0] - Success write worker group heartBeatInfo into registry, workGroupPath: [/nodes/worker/default/172.16.10.15:1234] workerHeartBeatInfo: {"startupTime":1665969146369,"reportTime":1665970039602,"cpuUsage":0.36,"memoryUsage":0.68,"loadAverage":2.78,"availablePhysicalMemorySize":20.04,"maxCpuloadAvg":16.0,"reservedMemory":0.3,"diskAvailable":34.09,"serverStatus":0,"processId":11664,"workerHostWeight":100,"workerWaitingTaskCount":0,"workerExecThreadCount":100}
[INFO] 2022-10-17 09:27:22.011 +0800 org.apache.dolphinscheduler.server.worker.processor.TaskDispatchProcessor:[98] - [WorkflowInstance-0][TaskInstance-0] - task execute request message: TaskDispatchCommand(super=BaseCommand(messageSenderAddress=172.16.10.15:5678, messageReceiverAddress=null, messageSendTime=1665970042008), taskExecutionContext=TaskExecutionContext(taskInstanceId=484, taskName=datax_test, firstSubmitTime=Mon Oct 17 09:27:21 CST 2022, startTime=null, taskType=DATAX, host=null, executePath=null, logPath=null, taskJson=null, processId=0, processDefineCode=7195111179040, processDefineVersion=2, appIds=null, processInstanceId=468, scheduleTime=null, globalParams=null, executorId=1, cmdTypeIfComplement=7, tenantCode=root, queue=default, processDefineId=0, projectId=0, projectCode=7193666667040, taskParams={"localParams":[],"resourceList":[],"customConfig":0,"dsType":"MYSQL","dataSource":1,"dtType":"MYSQL","dataTarget":1,"sql":"select id,name from ods.ods_jdx_site","targetTable":"ods_jdx_site_copy1","jobSpeedByte":0,"jobSpeedRecord":1000,"preStatements":[],"postStatements":[],"xms":1,"xmx":1}, envFile=null, environmentConfig=export HADOOP_HOME=/opt/soft/hadoop
export HADOOP_CONF_DIR=/opt/soft/hadoop/etc/hadoop
export SPARK_HOME1=/opt/soft/spark1
export SPARK_HOME2=/opt/soft/spark2
export PYTHON_HOME=/opt/anaconda3/envs/py37
export JAVA_HOME=/usr/lib/java/jdk
export HIVE_HOME=/opt/soft/hive
export FLINK_HOME=/opt/flink
export DATAX_HOME=/opt/soft/datax/bin/datax.py
export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$PYTHON_HOME/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_HOME:$PATH, definedParams=null, prepareParamsMap=null, taskAppId=null, taskTimeoutStrategy=null, taskTimeout=2147483647, workerGroup=default, delayTime=0, currentExecutionStatus=TaskExecutionStatus{code=0, desc='submit success'}, taskLogName=null, resourceParametersHelper=org.apache.dolphinscheduler.plugin.task.api.parameters.resource.ResourceParametersHelper@48f0b0ea, endTime=null, sqlTaskExecutionContext=null, k8sTaskExecutionContext=K8sTaskExecutionContext{configYaml='null'}, resources={}, varPool=null, dryRun=0, paramsMap={}, dataQualityTaskExecutionContext=org.apache.dolphinscheduler.plugin.task.api.DataQualityTaskExecutionContext@292b7f5, cpuQuota=-1, memoryMax=-1))
[INFO] 2022-10-17 09:27:22.012 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[90] - [WorkflowInstance-468][TaskInstance-484] - Set task logger name: taskAppId=TASK-20221017-7195111179040_2-468-484
[INFO] 2022-10-17 09:27:22.013 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[136] - [WorkflowInstance-468][TaskInstance-484] - Begin to pulling task
[INFO] 2022-10-17 09:27:22.015 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[164] - [WorkflowInstance-468][TaskInstance-484] - Begin to initialize task
[INFO] 2022-10-17 09:27:22.015 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[168] - [WorkflowInstance-468][TaskInstance-484] - Set task startTime: Mon Oct 17 09:27:22 CST 2022
[INFO] 2022-10-17 09:27:22.015 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[172] - [WorkflowInstance-468][TaskInstance-484] - Set task envFile: /opt/dolphinscheduler/worker-server/conf/dolphinscheduler_env.sh
[INFO] 2022-10-17 09:27:22.015 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[176] - [WorkflowInstance-468][TaskInstance-484] - Set task appId: 468_484
[INFO] 2022-10-17 09:27:22.015 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[178] - [WorkflowInstance-468][TaskInstance-484] - End initialize task
[INFO] 2022-10-17 09:27:22.016 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[184] - [WorkflowInstance-468][TaskInstance-484] - Set task status to TaskExecutionStatus{code=1, desc='running'}
[INFO] 2022-10-17 09:27:22.017 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[187] - [WorkflowInstance-468][TaskInstance-484] - TenantCode:root check success
[INFO] 2022-10-17 09:27:22.017 +0800 org.apache.dolphinscheduler.common.utils.FileUtils:[140] - [WorkflowInstance-468][TaskInstance-484] - create dir success /tmp/dolphinscheduler/exec/process/7193666667040/7195111179040_2/468/484
[INFO] 2022-10-17 09:27:22.017 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[190] - [WorkflowInstance-468][TaskInstance-484] - ProcessExecDir:/tmp/dolphinscheduler/exec/process/7193666667040/7195111179040_2/468/484 check success
[INFO] 2022-10-17 09:27:22.017 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[193] - [WorkflowInstance-468][TaskInstance-484] - Resources:{} check success
[INFO] 2022-10-17 09:27:22.017 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[203] - [WorkflowInstance-468][TaskInstance-484] - Task plugin: DATAX create success
[INFO] 2022-10-17 09:27:22.017 +0800 TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[134] - [WorkflowInstance-468][TaskInstance-484] - datax task params {"localParams":[],"resourceList":[],"customConfig":0,"dsType":"MYSQL","dataSource":1,"dtType":"MYSQL","dataTarget":1,"sql":"select id,name from ods_jdx_site","targetTable":"ods_jdx_site_copy","jobSpeedByte":0,"jobSpeedRecord":1000,"preStatements":[],"postStatements":[],"xms":1,"xmx":1}
[INFO] 2022-10-17 09:27:22.018 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[206] - [WorkflowInstance-468][TaskInstance-484] - Success initialized task plugin instance success
[INFO] 2022-10-17 09:27:22.018 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[209] - [WorkflowInstance-468][TaskInstance-484] - Success set taskVarPool: null
[ERROR] 2022-10-17 09:27:22.019 +0800 TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.datax.DataxTask:[169] - [WorkflowInstance-468][TaskInstance-484] - datax task error
java.lang.NullPointerException: null
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.addCustomParameters(DataxTask.java:426)
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.buildShellCommandFile(DataxTask.java:400)
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.handle(DataxTask.java:157)
        at org.apache.dolphinscheduler.server.worker.runner.DefaultWorkerDelayTaskExecuteRunnable.executeTask(DefaultWorkerDelayTaskExecuteRunnable.java:48)
        at org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable.run(WorkerTaskExecuteRunnable.java:151)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[ERROR] 2022-10-17 09:27:22.020 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[156] - [WorkflowInstance-468][TaskInstance-484] - Task execute failed, due to meet an exception
org.apache.dolphinscheduler.plugin.task.api.TaskException: Execute DataX task failed
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.handle(DataxTask.java:171)
        at org.apache.dolphinscheduler.server.worker.runner.DefaultWorkerDelayTaskExecuteRunnable.executeTask(DefaultWorkerDelayTaskExecuteRunnable.java:48)
        at org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable.run(WorkerTaskExecuteRunnable.java:151)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: null
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.addCustomParameters(DataxTask.java:426)
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.buildShellCommandFile(DataxTask.java:400)
        at org.apache.dolphinscheduler.plugin.task.datax.DataxTask.handle(DataxTask.java:157)
        ... 9 common frames omitted
[INFO] 2022-10-17 09:27:22.173 +0800 org.apache.dolphinscheduler.server.worker.processor.TaskExecuteRunningAckProcessor:[61] - [WorkflowInstance-0][TaskInstance-484] - task execute running ack command : TaskExecuteRunningAckMessage(success=true, taskInstanceId=484)
[INFO] 2022-10-17 09:27:23.021 +0800 org.apache.dolphinscheduler.service.log.LogClientService:[206] - [WorkflowInstance-468][TaskInstance-484] - Begin to get appIds from worker: 172.16.10.15:1234 taskLogPath: /opt/dolphinscheduler/worker-server/logs/20221017/7195111179040_2-468-484.log
[INFO] 2022-10-17 09:27:23.022 +0800 org.apache.dolphinscheduler.service.log.LogClientService:[223] - [WorkflowInstance-468][TaskInstance-484] - Get appIds: [] from worker: 172.16.10.15:1234 taskLogPath: /opt/dolphinscheduler/worker-server/logs/20221017/7195111179040_2-468-484.log
[INFO] 2022-10-17 09:27:23.024 +0800 org.apache.dolphinscheduler.service.log.LogClientService:[81] - [WorkflowInstance-468][TaskInstance-484] - logger client closed
[INFO] 2022-10-17 09:27:23.024 +0800 org.apache.dolphinscheduler.server.utils.ProcessUtils:[213] - [WorkflowInstance-468][TaskInstance-484] - The current appId is empty, don't need to kill the yarn job, taskInstanceId: 484
[INFO] 2022-10-17 09:27:23.025 +0800 TaskLogLogger-class org.apache.dolphinscheduler.server.worker.runner.WorkerTaskExecuteRunnable:[114] - [WorkflowInstance-468][TaskInstance-484] - Get a exception when execute the task, will send the task execute result to master, the current task execute result is TaskExecutionStatus{code=6, desc='failure'}
[INFO] 2022-10-17 09:27:23.168 +0800 org.apache.dolphinscheduler.server.worker.processor.TaskExecuteResultAckProcessor:[57] - [WorkflowInstance-0][TaskInstance-0] - task execute response ack command : TaskExecuteAckCommand(super=BaseCommand(messageSenderAddress=172.16.10.15:5678, messageReceiverAddress=172.16.10.15:1234, messageSendTime=1665970043167), taskInstanceId=484, success=true)
[INFO] 2022-10-17 09:27:29.623 +0800 org.apache.dolphinscheduler.server.worker.task.WorkerHeartBeatTask:[88] - [WorkflowInstance-0][TaskInstance-0] - Success write worker group heartBeatInfo into registry, workGroupPath: [/nodes/worker/default/172.16.10.15:1234] workerHeartBeatInfo: {"startupTime":1665969146369,"reportTime":1665970049614,"cpuUsage":0.36,"memoryUsage":0.68,"loadAverage":3.63,"availablePhysicalMemorySize":20.03,"maxCpuloadAvg":16.0,"reservedMemory":0.3,"diskAvailable":34.09,"serverStatus":0,"processId":11664,"workerHostWeight":100,"workerWaitingTaskCount":0,"workerExecThreadCount":100}

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java
@@ -135,7 +135,7 @@ public class DataxTask extends AbstractTask {
         logger.info("datax task params {}", taskExecutionContext.getTaskParams());
         dataXParameters = JSONUtils.parseObject(taskExecutionContext.getTaskParams(), DataxParameters.class);
 
-        if (!dataXParameters.checkParameters()) {
+        if (dataXParameters == null || !dataXParameters.checkParameters()) {
             throw new RuntimeException("datax task params is not valid");
         }
 
@@ -145,7 +145,7 @@ public class DataxTask extends AbstractTask {
     /**
      * run DataX process
      *
-     * @throws Exception if error throws Exception
+     * @throws TaskException if error throws Exception
      */
     @Override
     public void handle(TaskCallBack taskCallBack) throws TaskException {
@@ -153,7 +153,7 @@ public class DataxTask extends AbstractTask {
             // replace placeholder,and combine local and global parameters
             Map<String, Property> paramsMap = taskExecutionContext.getPrepareParamsMap();
 
-            // run datax procesDataSourceService.s
+            // run datax processDataSourceService
             String jsonFilePath = buildDataxJsonFile(paramsMap);
             String shellCommandFilePath = buildShellCommandFile(jsonFilePath, paramsMap);
             TaskResponse commandExecuteResult = shellCommandExecutor.run(shellCommandFilePath);
@@ -233,7 +233,6 @@ public class DataxTask extends AbstractTask {
      * build datax job config
      *
      * @return collection of datax job config JSONObject
-     * @throws SQLException if error throws SQLException
      */
     private List<ObjectNode> buildDataxJobContentJson() {
 
@@ -393,18 +392,17 @@ public class DataxTask extends AbstractTask {
         }
 
         // datax python command
-        StringBuilder sbr = new StringBuilder();
-        sbr.append(getPythonCommand());
-        sbr.append(" ");
-        sbr.append(DATAX_PATH);
-        sbr.append(" ");
-        sbr.append(loadJvmEnv(dataXParameters));
-        sbr.append(addCustomParameters(paramsMap));
-        sbr.append(" ");
-        sbr.append(jobConfigFilePath);
+        String sbr = getPythonCommand() +
+                " " +
+                DATAX_PATH +
+                " " +
+                loadJvmEnv(dataXParameters) +
+                addCustomParameters(paramsMap) +
+                " " +
+                jobConfigFilePath;
 
         // replace placeholder
-        String dataxCommand = ParameterUtils.convertParameterPlaceholders(sbr.toString(), ParamUtils.convert(paramsMap));
+        String dataxCommand = ParameterUtils.convertParameterPlaceholders(sbr, ParamUtils.convert(paramsMap));
 
         logger.debug("raw script : {}", dataxCommand);
 
@@ -424,10 +422,14 @@ public class DataxTask extends AbstractTask {
     }
 
     private StringBuilder addCustomParameters(Map<String, Property> paramsMap) {
-        StringBuilder customParameters = new StringBuilder("-p\"");
+        if (paramsMap == null || paramsMap.size() == 0) {
+            return new StringBuilder();
+        }
+        StringBuilder customParameters = new StringBuilder("-p \"");
         for (Map.Entry<String, Property> entry : paramsMap.entrySet()) {
             customParameters.append(String.format(CUSTOM_PARAM, entry.getKey(), entry.getValue().getValue()));
         }
+        customParameters.replace(4, 5, "");
         customParameters.append("\"");
         return customParameters;
     }
@@ -530,12 +532,12 @@ public class DataxTask extends AbstractTask {
                     }
                 } else {
                     throw new RuntimeException(
-                            String.format("grammatical analysis sql column [ %s ] failed", item.toString()));
+                            String.format("grammatical analysis sql column [ %s ] failed", item));
                 }
 
                 if (columnName == null) {
                     throw new RuntimeException(
-                            String.format("grammatical analysis sql column [ %s ] failed", item.toString()));
+                            String.format("grammatical analysis sql column [ %s ] failed", item));
                 }
 
                 columnNames[i] = columnName;

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/test/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTaskTest.java
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/test/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTaskTest.java
@@ -1,0 +1,327 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.task.datax;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.dolphinscheduler.common.utils.FileUtils;
+import org.apache.dolphinscheduler.plugin.datasource.api.plugin.DataSourceClientProvider;
+import org.apache.dolphinscheduler.plugin.task.api.ShellCommandExecutor;
+import org.apache.dolphinscheduler.plugin.task.api.TaskCallBack;
+import org.apache.dolphinscheduler.plugin.task.api.TaskException;
+import org.apache.dolphinscheduler.plugin.task.api.TaskExecutionContext;
+import org.apache.dolphinscheduler.plugin.task.api.enums.DataType;
+import org.apache.dolphinscheduler.plugin.task.api.enums.Direct;
+import org.apache.dolphinscheduler.plugin.task.api.enums.TaskRunStatus;
+import org.apache.dolphinscheduler.plugin.task.api.model.Property;
+import org.apache.dolphinscheduler.plugin.task.api.model.TaskResponse;
+import org.apache.dolphinscheduler.plugin.task.api.parameters.resource.ResourceParametersHelper;
+import org.apache.dolphinscheduler.spi.datasource.BaseConnectionParam;
+import org.apache.dolphinscheduler.spi.enums.DbType;
+import org.apache.dolphinscheduler.spi.utils.JSONUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+public class DataxTaskTest {
+
+    private DataxTask dataxTask;
+
+    private final TaskCallBack taskCallBack = (taskInstanceId, appIds) -> {};
+
+    @BeforeEach
+    public void before() throws Exception {
+        TaskExecutionContext taskExecutionContext = mock(TaskExecutionContext.class);
+        ResourceParametersHelper resourceParametersHelper = new ResourceParametersHelper();
+        String parameters = JSONUtils.toJsonString(createDataxParameters());
+        when(taskExecutionContext.getTaskParams()).thenReturn(parameters);
+        taskExecutionContext.setResourceParametersHelper(resourceParametersHelper);
+        this.dataxTask = new DataxTask(taskExecutionContext);
+        this.dataxTask.init();
+    }
+
+    @Test
+    public void testHandleNullParamsMap() throws Exception {
+        String parameters = JSONUtils.toJsonString(createDataxParameters());
+        TaskExecutionContext taskExecutionContext = buildTestTaskExecutionContext();
+        taskExecutionContext.setPrepareParamsMap(null);
+        taskExecutionContext.setTaskParams(parameters);
+        DataxTask dataxTask = new DataxTask(taskExecutionContext);
+        dataxTask.init();
+
+        ShellCommandExecutor shellCommandExecutor = mock(ShellCommandExecutor.class);
+        Field shellCommandExecutorFiled = DataxTask.class.getDeclaredField("shellCommandExecutor");
+        shellCommandExecutorFiled.setAccessible(true);
+        shellCommandExecutorFiled.set(dataxTask, shellCommandExecutor);
+
+        TaskResponse taskResponse = new TaskResponse();
+        taskResponse.setStatus(TaskRunStatus.SUCCESS);
+        taskResponse.setExitStatusCode(0);
+        taskResponse.setProcessId(1);
+        when(shellCommandExecutor.run(anyString())).thenReturn(taskResponse);
+
+        dataxTask.handle(taskCallBack);
+        Assertions.assertEquals(0, dataxTask.getExitStatusCode());
+
+        File jsonFile = new File("/tmp/execution/app-id_job.json");
+        InputStream json = Files.newInputStream(jsonFile.toPath());
+        String resultStr = FileUtils.readFile2Str(json);
+        Assertions.assertEquals(resultStr, getJsonString());
+        boolean delete = jsonFile.delete();
+        Assertions.assertTrue(delete);
+
+        File shellCommandFile = SystemUtils.IS_OS_WINDOWS ?
+                new File("/tmp/execution/app-id_node.bat") :
+                new File("/tmp/execution/app-id_node.sh");
+        InputStream shellCommandInputStream = Files.newInputStream(shellCommandFile.toPath());
+        String shellCommandStr = FileUtils.readFile2Str(shellCommandInputStream);
+        Assertions.assertEquals(shellCommandStr, "python2.7 ${DATAX_HOME}/bin/datax.py  --jvm=\"-Xms1G -Xmx1G\" " +
+                " /tmp/execution/app-id_job.json");
+        delete = shellCommandFile.delete();
+        Assertions.assertTrue(delete);
+    }
+
+    @Test
+    public void testHandleParamsMap() throws Exception {
+        String parameters = JSONUtils.toJsonString(createDataxParameters());
+        TaskExecutionContext taskExecutionContext = buildTestTaskExecutionContext();
+
+        taskExecutionContext.setPrepareParamsMap(createPrepareParamsMap());
+        taskExecutionContext.setTaskParams(parameters);
+        DataxTask dataxTask = new DataxTask(taskExecutionContext);
+        dataxTask.init();
+
+        ShellCommandExecutor shellCommandExecutor = mock(ShellCommandExecutor.class);
+        Field shellCommandExecutorFiled = DataxTask.class.getDeclaredField("shellCommandExecutor");
+        shellCommandExecutorFiled.setAccessible(true);
+        shellCommandExecutorFiled.set(dataxTask, shellCommandExecutor);
+
+        TaskResponse taskResponse = new TaskResponse();
+        taskResponse.setStatus(TaskRunStatus.SUCCESS);
+        taskResponse.setExitStatusCode(0);
+        taskResponse.setProcessId(1);
+        when(shellCommandExecutor.run(anyString())).thenReturn(taskResponse);
+
+        dataxTask.handle(taskCallBack);
+        Assertions.assertEquals(0, dataxTask.getExitStatusCode());
+
+        File jsonFile = new File("/tmp/execution/app-id_job.json");
+        InputStream json = Files.newInputStream(jsonFile.toPath());
+        String resultStr = FileUtils.readFile2Str(json);
+        Assertions.assertEquals(resultStr, getJsonString());
+        boolean delete = jsonFile.delete();
+        Assertions.assertTrue(delete);
+
+        File shellCommandFile = SystemUtils.IS_OS_WINDOWS ?
+                new File("/tmp/execution/app-id_node.bat") :
+                new File("/tmp/execution/app-id_node.sh");
+        InputStream shellCommandInputStream = Files.newInputStream(shellCommandFile.toPath());
+        String shellCommandStr = FileUtils.readFile2Str(shellCommandInputStream);
+        Assertions.assertEquals(shellCommandStr, "python2.7 ${DATAX_HOME}/bin/datax.py  --jvm=\"-Xms1G -Xmx1G\" " +
+                "-p \"-DDT=DT -DDS=DS\" /tmp/execution/app-id_job.json");
+        delete = shellCommandFile.delete();
+        Assertions.assertTrue(delete);
+    }
+
+    @Test
+    public void testHandleInterruptedException() throws Exception {
+        String parameters = JSONUtils.toJsonString(createDataxParameters());
+        TaskExecutionContext taskExecutionContext = buildTestTaskExecutionContext();
+        taskExecutionContext.setPrepareParamsMap(null);
+        taskExecutionContext.setTaskParams(parameters);
+        DataxTask dataxTask = new DataxTask(taskExecutionContext);
+        dataxTask.init();
+
+        ShellCommandExecutor shellCommandExecutor = mock(ShellCommandExecutor.class);
+        Field shellCommandExecutorFiled = DataxTask.class.getDeclaredField("shellCommandExecutor");
+        shellCommandExecutorFiled.setAccessible(true);
+        shellCommandExecutorFiled.set(dataxTask, shellCommandExecutor);
+
+        when(shellCommandExecutor.run(anyString())).thenThrow(new InterruptedException("Command execution failed"));
+        Assertions.assertThrows(TaskException.class, () -> dataxTask.handle(taskCallBack));
+    }
+
+    @Test
+    public void testHandleIOException() throws Exception {
+        String parameters = JSONUtils.toJsonString(createDataxParameters());
+        TaskExecutionContext taskExecutionContext = buildTestTaskExecutionContext();
+        taskExecutionContext.setPrepareParamsMap(null);
+        taskExecutionContext.setTaskParams(parameters);
+        DataxTask dataxTask = new DataxTask(taskExecutionContext);
+        dataxTask.init();
+
+        ShellCommandExecutor shellCommandExecutor = mock(ShellCommandExecutor.class);
+        Field shellCommandExecutorFiled = DataxTask.class.getDeclaredField("shellCommandExecutor");
+        shellCommandExecutorFiled.setAccessible(true);
+        shellCommandExecutorFiled.set(dataxTask, shellCommandExecutor);
+
+        when(shellCommandExecutor.run(anyString())).thenThrow(new IOException("Command execution failed"));
+        Assertions.assertThrows(TaskException.class, () -> dataxTask.handle(taskCallBack));
+    }
+
+    @Test
+    public void testTryExecuteSqlResolveColumnNames() throws Exception {
+        BaseConnectionParam baseConnectionParam = mock(BaseConnectionParam.class);
+        try (
+                MockedStatic<DataSourceClientProvider> mockedStaticDataSourceClientProvider =
+                        mockStatic(DataSourceClientProvider.class)) {
+            DataSourceClientProvider clientProvider = mock(DataSourceClientProvider.class);
+            when(DataSourceClientProvider.getInstance()).thenReturn(clientProvider);
+            mockedStaticDataSourceClientProvider.when(DataSourceClientProvider::getInstance).thenReturn(clientProvider);
+
+            Connection connection = mock(Connection.class);
+            when(clientProvider.getConnection(Mockito.any(), Mockito.any())).thenReturn(connection);
+
+            PreparedStatement stmt = mock(PreparedStatement.class);
+            when(connection.prepareStatement(anyString())).thenReturn(stmt);
+
+            ResultSetMetaData md = mock(ResultSetMetaData.class);
+            when(md.getColumnCount()).thenReturn(1);
+            when(md.getColumnName(eq(1))).thenReturn("something");
+
+            ResultSet resultSet = mock(ResultSet.class);
+            when(resultSet.getMetaData()).thenReturn(md);
+            when(stmt.executeQuery()).thenReturn(resultSet);
+
+            String[] rows = this.dataxTask.tryExecuteSqlResolveColumnNames(DbType.MYSQL,baseConnectionParam, "");
+            Assertions.assertEquals(rows.length, 1);
+            Assertions.assertEquals(rows[0], "something");
+
+            when(connection.prepareStatement(anyString())).thenThrow(new SQLException("Connection failed"));
+            String[] nullRows = this.dataxTask.tryExecuteSqlResolveColumnNames(DbType.MYSQL,baseConnectionParam, "");
+            Assertions.assertNull(nullRows);
+        }
+    }
+
+    @Test
+    public void testGetPythonCommand() {
+        Assertions.assertEquals(dataxTask.getPythonCommand(""), "python2.7");
+        Assertions.assertEquals(dataxTask.getPythonCommand("/bin/python"), "/bin/python2.7");
+
+        String pythonCommand = dataxTask.getPythonCommand("/opt/python");
+        pythonCommand = pythonCommand.replace(File.separator, "/");
+        Assertions.assertEquals(pythonCommand, "/opt/python/bin/python2.7");
+    }
+
+    @Test
+    public void testLoadJvmEnv() {
+        DataxParameters dataXParameters = createDataxParameters();
+        dataXParameters.setXms(3);
+        dataXParameters.setXmx(4);
+        Assertions.assertEquals(dataxTask.loadJvmEnv(dataXParameters), " --jvm=\"-Xms3G -Xmx4G\" ");
+    }
+
+    private DataxParameters createDataxParameters() {
+        DataxParameters dataxParameters = new DataxParameters();
+        dataxParameters.setCustomConfig(1);
+        dataxParameters.setDsType("mysql");
+        dataxParameters.setDataSource(1);
+        dataxParameters.setJson(getJsonString());
+        dataxParameters.setDataTarget(2);
+        dataxParameters.setSql("SELECT count(*) FROM table");
+        dataxParameters.setTargetTable("user.name");
+        return dataxParameters;
+    }
+
+    private Map<String, Property> createPrepareParamsMap() {
+        Map<String, Property> paramsMap = new HashMap<>();
+        Property dtProperty = new Property();
+        dtProperty.setProp("DT");
+        dtProperty.setDirect(Direct.IN);
+        dtProperty.setType(DataType.VARCHAR);
+        dtProperty.setValue("DT");
+        Property dsProperty = new Property();
+        dsProperty.setProp("DS");
+        dsProperty.setDirect(Direct.IN);
+        dsProperty.setType(DataType.VARCHAR);
+        dsProperty.setValue("DS");
+        paramsMap.put("DT", dtProperty);
+        paramsMap.put("DS", dsProperty);
+        return paramsMap;
+    }
+
+    private TaskExecutionContext buildTestTaskExecutionContext() {
+        TaskExecutionContext taskExecutionContext = new TaskExecutionContext();
+        taskExecutionContext.setTaskAppId("app-id");
+        taskExecutionContext.setExecutePath("/tmp/execution");
+        return taskExecutionContext;
+    }
+
+    private String getJsonString() {
+        return "{\n" +
+                "  \"job\": {\n" +
+                "    \"content\": [\n" +
+                "      {\n" +
+                "        \"reader\": {\n" +
+                "          \"name\": \"stream reader\",\n" +
+                "          \"parameter\": {\n" +
+                "            \"sliceRecordCount\": 10,\n" +
+                "            \"column\": [\n" +
+                "              {\n" +
+                "                \"type\": \"long\",\n" +
+                "                \"value\": \"10\"\n" +
+                "              },\n" +
+                "              {\n" +
+                "                \"type\": \"string\",\n" +
+                "                \"value\": \"Hello DataX\"\n" +
+                "              }\n" +
+                "            ]\n" +
+                "          }\n" +
+                "        },\n" +
+                "        \"writer\": {\n" +
+                "          \"name\": \"stream writer\",\n" +
+                "          \"parameter\": {\n" +
+                "            \"encoding\": \"UTF-8\",\n" +
+                "            \"print\": true\n" +
+                "          }\n" +
+                "        }\n" +
+                "      }\n" +
+                "    ],\n" +
+                "    \"setting\": {\n" +
+                "      \"speed\": {\n" +
+                "        \"channel\": 5\n" +
+                "       }\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+    }
+}

